### DISCUSSION TITLE
migrating HA

### DISCUSSION THREAD

**Discussion by userMak** [krahabb/meross_lan#216] [General]
Hi all
I had merross lan set up in my NUC with Ubuntu 18.04 LTS and HA supervised. Today I changed to Hassos. I restored the back up file and the merros lan integration is red
![image](https://user-images.githubusercontent.com/55662933/192259614-c2063355-b0ec-4cef-ac8a-d3153382279e.png)

I reconfigured the devices with npx etc and now HA has discovered all the devices and I have to reconnect them
![image](https://user-images.githubusercontent.com/55662933/192259845-75e1b6c8-36b9-499a-a48d-beb5e36a253b.png)
in the configuration I have the following and I am not sure what option to choose
user set, hack mode, or cloud retrieve. 
1. Can someone help please? 
2. It would be better to completely remove the merros lan integration and re add it or I should remove the old entities/devices which are not available (red)

![image](https://user-images.githubusercontent.com/55662933/192259956-6dbb36b7-723e-4471-b719-f1c0e28455d6.png)

edit: I just noticed that my old devices have different setup page/options
![image](https://user-images.githubusercontent.com/55662933/192263015-e49b42aa-ff47-4941-a82c-f86c6bd9a0a7.png)

**Comment by userMak** [comment]
I am completely lost here. Some help would be usefull please.
I have connected all my devices with mqtt as usual (like the following)
node meross setup --wifi-ssid xxx --wifi-pass xxx --mqtt mqtts://192.168.1.xx

I can not finf a way to add them in HA. I migrate the entire system easily and I am stuck here - almost all the lights in the house are missing. What should I do? And why is asking me for the meross app account credentials?

**Reply by nao-pon** [reply]
It is necessary to create a user in HA using the Credentials displayed by the setup command. is it done?

**Reply by nao-pon** [reply]
Once the mqtt connection is complete, the device should be recognized by auto discovery.

**Comment by userMak** [comment]
I have several users like this
![image](https://user-images.githubusercontent.com/55662933/192309003-fb10126e-bd47-4289-831c-1dc0b23f1d19.png)

I reconfigured mosquito and just now I manage to add one light. I used the hack method all the others didn't work. But it has changed the name of the entity and I am trying to figure out how I could keep the old names (in order to not have to change all my automations)

**Reply by nao-pon** [reply]
Also, HA core 2022.9.6 has problems with MQTT connections. See thread #213.

**Comment by userMak** [comment]
As I see, it install the light as outlet.
When I tried to change the device name as the older ones, it made a hidden switch and a light switch.

![image](https://user-images.githubusercontent.com/55662933/192312128-74434cbc-17aa-4806-aaec-9f6c60015071.png)

**Comment by userMak** [comment]
this is how they first installed (this is a double switch)
![image](https://user-images.githubusercontent.com/55662933/192313128-20b1cf69-9cdd-4107-ad17-a900aa731bad.png)

**Comment by krahabb** [comment]
Hello @userMak ,
the HA core team is introducing a new naming scheme (even if they said the change should be backward compatible in some way - at least for some time/version) so the issue might be due to an 'unstable' implementation.

Also, the 'Hack mode' is needed when you want to enter a null key (which in the beginning was the default option with no need for credentail access. Now, the configuration UI is more oriented towards people which are still using the Meross cloud together with the HTTP protocol which, I think, are the majority of meross_lan users).

Trying to recap a bit of your configuration:
You likely installed/configured the old ones through MQTT but, when you installed the new HA version, the MQTT was not working anymore so the devices were 'unknown' to the system and it re-discovered them through the HTTP interface.
Now, if your devices are unbinded from the meross cloud and you configured them through 'node meross setup...' with a blank password you could still use them with HTTP (meross_lan automatically switches between HTTP <-> MQTT when it detects the protocol is working) by entering their adrress/IP and the blank password (with Hack-Mode). This way you'll have to reconfigure them all of course. 
Sadly, when a device is initially configured through MQTT, you cannot edit/enter its IP address so, in order to have it work again you need to either fix the communication with the MQTT broker or reconfigure meross_lan from scratch by deleting/re-adding.

If you just want to 'recover' the old configuration(s) I guess you could achieve that by using the fix proposed by @nao-pon in #213 until I release an 'official' fix

**Comment by userMak** [comment]
Hi @krahabb 
I did reconfigure all the devices, I also changed all their names with the old ones in order to keep my automations stable. I hope that I won't need to do anything more. (although I am not so optimistic)

**Reply by krahabb** [reply]
I think the issue was really a consequence of the 2022.9.6 mqtt behaviour change. meross_lan was not able to talk mqtt anymore so the system just (re)discovered devices through HTTP.
I'll try a better way to unify the configuration between devices discovered over HTTP vs MQTT so to be able to handle this kind of issues in the future. Actual choices are a stratification of the component evolution over time so they could likely be improved a bit to 'smooth' out similar issues.
Not sure tho..this will need some deep thinking ;)
